### ISSUE TITLE
Proposal: DiffLog in CanvasBufferAPI

### ISSUE THREAD

**Issue by skyzh** [untitled-group/kanbasu#30]
In kanbasu, we would like to let our users know which pages have been changed. This can be done with `DiffLog`.

`DiffLog` is a set of keys in KvStore with prefix `kanbasu/api_diff/`. When APIs are called within `CanvasBufferAPI`, it will first get data from KvStore, then from REST API, as usual. After we implement `DiffLog`, data from two endpoints are compared. If there is any difference, a key will be created.

For example, we call `api.getCourse(23333)`. We get a course object from KvStore, then from REST API. After that, we found that two objects are not exactly the same. Then, we add a key at `kanbasu/api_diff/courses/by_id/23333` with value `diff_detected_at`. Then in our UI components, we could display a red dot to indicate there's some changes.

Then here comes another issue. What if our user is just viewing the UI of course 23333. The `DiffLog` is generated after the stream ends. Only after the UI has been rendered do we observe the content has changed. Therefore, we should add a new parameter to all API calls: `is_current_view`. If the API called is content of current view, we do not add this entry into `DiffLog`.
